src/array.ts

```diff
@@ -219,6 +221,29 @@ Array.prototype.asyncFilter = async function <T>(
   return this.filter((_v, index) => results[index]);
 };
 
+Array.prototype.asyncSome = async function <T>(
+  this: Array<T>,
+  predicate: (v: T) => boolean | Promise<boolean>,
+): Promise<boolean> {
+  const asyncResults: Promise<boolean>[] = [];
+  for (const el of this) {
+    const result = predicate(el);
+    if (typeof result === "boolean") {
+      if (result) {
+        return true;
+      }
+    } else {
+      asyncResults.push(result);
+    }
+  }
+  for (const result of asyncResults) {
```

Comment:
Hmm. I can't find a way to do it, so I guess it's not possible, but it would be nice to be able to cancel pending promises if we return early.

But I guess AbortController is a feature of browser `fetch`, not of Promises. So, can't really cancel pending promises.